app/helpers/ucsc_thumbnail_helper.rb

```diff
@@ -1,15 +1,20 @@
 module UcscThumbnailHelper
 
   def square_thumbnail_region(doc, size="150,")
-    doc = SolrDocument.find(doc) if doc.is_a?(String)
-    size = "#{size}," unless (size.is_a?(String) && size.include?(','))
-    hh = Array(doc.height).first.to_i
-    ww = Array(doc.width).first.to_i
-    if hh > ww
-      square_region = "0,#{(hh-ww)/2},#{ww},#{ww}"
-    else
-      square_region =  "#{(ww-hh)/2},0,#{hh},#{hh}"
-    end
+    # commenting out the code to calculate thumbnail region
+    # Below code causes errors when the height and width are huge.
+    # TODO : fix this in Fedora - see why image region is huge
+    # For now using square region for all thumbnails
+    # doc = SolrDocument.find(doc) if doc.is_a?(String)
+    # size = "#{size}," unless (size.is_a?(String) && size.include?(','))
+    # hh = Array(doc.height).first.to_i
+    # ww = Array(doc.width).first.to_i
+    # if hh > ww
+    #   square_region = "0,#{(hh-ww)/2},#{ww},#{ww}"
+    # else
+    #   square_region =  "#{(ww-hh)/2},0,#{hh},#{hh}"
+    # end
+    square_region = "square"
```

Comment:
Can this happen only as a fallback instead of all the time? For example when the width is > 5000 or something like that?

Reply:
It can be done..but I thought that it would be better if all the thumbnail urls are uniform. If we go with the fallback approach..we need  threshold values for height and width.

Reply:
If we do it for all thumbnails we are more likely to mess up existing thumbnails for non-square images, like the team was concerned about in standup today. With a threshhold for squares we can try to limit it to these problematic aerials. I'd say something around 5000 for width and height would likely limit it to just these aerials with problems, but I leave that to you to decide the exact number based on what you saw.

Reply:
Makes sense. Let me test a few things and then update the code.